pkg/store/metrics/metrics.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package metrics
+
+import (
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/thanos-io/thanos/pkg/store/storepb"
+)
+
+// SeriesQueryPerformanceMetricsAggregator aggregates results from fanned-out queries into a histogram given their
+// response's shape.
+type SeriesQueryPerformanceMetricsAggregator struct {
```

Comment:
is `SeriesQueryPerformanceMetricsAggregator` being used outside the `metrics` pkg? If not, I think we could leave it unexported and use the exported constructor `NewSeriesQueryPerformanceMetricsAggregator`